### PR TITLE
feat(core): introduce decimal type fhir adapter

### PR DESCRIPTION
**Pull request by ngouy**
## What is this about

Introduce adapter for the [fhir decimal type](https://hl7.org/fhir/datatypes.html#decimal)

### CODE REVIEW

File: packages/core/r4b/data-types/decimal.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * Return a {@link FhirDecimalTypeAdapter} that uses the [`Intl` API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)
+ * (ECMAScript Internationalization API)
+ * @param locale - see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locales_argument
+ */
+export function fhirDecimalTypeAdapter(
+  locale: string | undefined
+): FhirDecimalTypeAdapter {
+  // JIT locale check
+  Intl.DateTimeFormat(locale);
```
Comment (julienblin) [inline packages/core/r4b/data-types/decimal.ts]: Nitpicking...
```suggestion
  Intl.NumberFormat(locale);
```

File: packages/core/r4b/data-types/decimal.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      if (!value?.trim()) {
+        return undefined;
+      }
+
+      if (!value.trim().match(fhirDecimalRegexp))
+        throw new Error(
+          "Value does not match the fhir date format as described in `https://hl7.org/fhir/datatypes.html#decimal'"
+        );
+
+      if (value.includes(".")) return parseFloat(value);
```
Comment (julienblin) [inline packages/core/r4b/data-types/decimal.ts]: We may run into issues related to precision here, specifically with those horrible IEEE754 floats.
Maybe we should parse it into https://mikemcl.github.io/decimal.js/?

**Review by julienblin — CHANGES_REQUESTED** [review]
Let's discuss when you have some time

File: packages/core/r4b/data-types/date.ts [hunk 3]
```diff
@@ -105,25 +110,26 @@ export function fhirDateTypeAdapter(
...
-          return new Intl.DateTimeFormat(locale, {
-            dateStyle: options?.dateStyle || undefined,
-          }).format(fhirDate.date);
+          options.dateStyle = style || undefined;
+          break;
         default:
           throw new Error(`Unknown date flavour ${fhirDate.flavour}`);
       }
+
+      return new Intl.DateTimeFormat(locale, options).format(fhirDate.date);
```
Comment (ngouy) [inline packages/core/r4b/data-types/date.ts]: @julienblin it was on my stash

I can roll it back

Which way should we favor?

**Review by julienblin — APPROVED** [review]
LGTM! - I will just squash the commits and merge (to avoid too much noise on Changelog).

Thank you! 🙏

File: packages/core/r4b/data-types/date.ts (hunk 3 above)
Comment (julienblin) [reply packages/core/r4b/data-types/date.ts]: We'll keep your version :-)

**Pull request merged**